extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php

```diff
@@ -13,14 +13,12 @@ static public function onBeforePageDisplay( OutputPage $out, Skin $skin ) {
 
 	static public function onImageServingCollectImages( &$imageNamesArray, $articleTitle ) {
 		if ( $articleTitle ) {
-
-			$dataService = new PortableInfoboxDataService();
-			$infoboxData = $dataService->getInfoboxDataByTitle( $articleTitle );
-			$infoboxImages = $dataService->getImageListFromInfoboxesData( $infoboxData );
+			$infoboxImages = ( new PortableInfoboxDataService( $articleTitle ) )->getImages();
```

Comment:
I think that the PortableInfoboxDataService building API has become confusing. Could we use newFromPageId and newFromTitle instead of current newFromPageId and __construct?